auth/api/iam/api.go

```diff
@@ -118,30 +116,21 @@ func (r Wrapper) HandleTokenRequest(ctx context.Context, request HandleTokenRequ
 		// Options:
 		// - OpenID4VCI
 		// - OpenID4VP, vp_token is sent in Token Response
+		panic("not implemented")
 	case "vp_token":
 		// Options:
 		// - service-to-service vp_token flow
+		panic("not implemented")
 	case "urn:ietf:params:oauth:grant-type:pre-authorized_code":
 		// Options:
 		// - OpenID4VCI
+		panic("not implemented")
 	default:
-		// TODO: Don't use openid4vci package for errors
-		return nil, openid4vci.Error{
-			Code:       openid4vci.InvalidRequest,
-			StatusCode: http.StatusBadRequest,
-			//Description: "invalid grant type",
+		return nil, OAuth2Error{
+			Code:        InvalidRequest,
+			Description: "unsupported grant_type",
```

Comment:
`invalid_request` or `unsupported_grant_type`? (not both)

Reply:
`unsupported_grant_type`